### ISSUE TITLE
`UselessCallOnNotNull` rule returns false positive when using the spread operator

### ISSUE THREAD

**Issue by luanpotter** [detekt/detekt#4236]
## Expected Behavior

Following code should be fine:

```kotlin
    val a = arrayOf(null, 1, 2, null)
    println(listOfNotNull(*a))
```

`a` contains null elements that I want to remove.

## Observed Behavior

However, detekt says:

```
        UselessCallOnNotNull - [main] at /home/luan/projects/detekt-testbed/src/main/kotlin/xyz/luan/test/Main.kt:5:13
```

## Steps to Reproduce

Please see [my repo](https://github.com/luanpotter/detekt-testbed/tree/luan.UselessCallOnNotNull) for a fully reproducible example (just run `./gradlew detektMain`). Or run detekt on the code above. I am using the latest 1.19.0-RC1 but seems to happen on 1.18 as well.

## Context

An example is this life function I use on my projects:

```kotlin
fun <T : Any> setOfNotNull(vararg elements: T?): Set<T> {
    return listOfNotNull(*elements).toSet()
}
```

I definitely **cannot** replace `listOfNotNull` there with `listOf`; yet detekt complains, which makes absolutely no sense to me.

## Your Environment

* Version of detekt used: 1.19.0-RC1
* Version of Gradle used (if applicable): 6.5
* Gradle scan link (add `--scan` option when running the gradle task): https://scans.gradle.com/s/xjnfqtmbct3yy
* Operating System and version: Arch Linux (GNU Linux 5.14.14-arch1-1)
* Link to your project (if it's a public repository): https://github.com/luanpotter/detekt-testbed/tree/luan.UselessCallOnNotNull
